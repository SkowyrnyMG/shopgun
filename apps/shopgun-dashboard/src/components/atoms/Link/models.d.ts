export interface LinkProps {
  url: string;
}
