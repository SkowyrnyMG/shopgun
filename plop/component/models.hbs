{{#insertOption typeOptions "interface"}}
export interface {{properCase name}}Props {

};
{{/insertOption}}

{{#insertOption typeOptions "namespace"}}
{{lowerCase namespaceOption}} namespace {{properCase name}}Type {

};
{{/insertOption}}