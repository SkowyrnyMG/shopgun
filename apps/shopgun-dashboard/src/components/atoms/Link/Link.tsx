import React, { FC } from "react";
import NextLink from "next/link";

import { LinkProps } from "./models.d";

import link from "./Link.module.scss";

const Link: FC<LinkProps> = ({ children, url }) => (
  <NextLink href={url}>
    <span className={link.wrapper}>{children}</span>
  </NextLink>
);

export default Link;
