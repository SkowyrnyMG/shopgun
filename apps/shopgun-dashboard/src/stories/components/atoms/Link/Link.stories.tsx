import { ComponentMeta, ComponentStory } from "@storybook/react";
import Link from "components/atoms/Link";

export default {
  title: "Link",
  component: Link,
  argTypes: {},
} as ComponentMeta<typeof Link>;

const Template: ComponentStory<typeof Link> = (args) => <Link {...args} />;

export const LinkVariant = Template.bind({});
