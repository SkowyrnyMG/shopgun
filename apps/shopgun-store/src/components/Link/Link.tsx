import React, { FC } from "react";
import NextLink from "next/link";

import link from "./Link.module.scss";

interface LinkProperties {
  url: string;
}

const Link: FC<LinkProperties> = ({ children, url }) => (
  <NextLink href={url}>
    <span className={link.link}>{children}</span>
  </NextLink>
);

export default Link;
