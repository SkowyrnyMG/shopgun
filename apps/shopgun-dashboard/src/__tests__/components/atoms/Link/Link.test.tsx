import React from "react";

import Link from "components/atoms/Link";
import { LinkProps } from "components/atoms/Link/models.d";
import { render, screen } from "testUtils";

const props: LinkProps = {
  url: "/about",
};

describe("Link - should pass all test cases", () => {
  test("the Link component renders correctly with required and optional props", () => {
    render(<Link {...props}>test</Link>);

    const linkNode = screen.getByText("test");

    // the component has been rendered
    expect(linkNode).toBeInTheDocument();

    // generating a snapshot
    expect(linkNode).toMatchSnapshot();
  });
});
