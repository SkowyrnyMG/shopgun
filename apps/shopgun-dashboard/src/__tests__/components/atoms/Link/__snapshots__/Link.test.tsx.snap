// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link - should pass all test cases the Link component renders correctly with required and optional props 1`] = `
<span
  class="wrapper"
>
  test
</span>
`;
